### ISSUE TITLE
Use underscore separation for large integers

### ISSUE THREAD

**Issue by mtaube** [realm/SwiftLint#924]
Any number greater than 4 characters should use ``_`` for separation.

```swift
// bad
let num = 10000000

// good
let num = 10_000_000
```

**Comment by jpsim** [comment]
Good idea!

**Comment by marcelofabri** [comment]
Maybe it should validate that the `_` are separating blocks of 3 digits?

Example from the [docs](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html#//apple_ref/doc/uid/TP40014097-CH5-ID309):
```swift
let oneMillion = 1_000_000
let justOverOneMillion = 1_000_000.000_000_1
```
